tests/Aspire.Hosting.Tests/ManifestGenerationTests.cs

```diff
@@ -721,6 +721,15 @@ public void VerifyTestProgramFullManifest()
                     }
                   }
                 },
+                "postgres-username": {
+                  "type": "parameter.v0",
+                  "value": "{postgres-username.inputs.value}",
+                  "inputs": {
+                    "value": {
+                      "type": "string"
+                    }
+                  }
+                },
```

Comment:
Does this mean that users will always be prompted for a user name when they deploy via `azd`?

Reply:
Is there a way to have the default value be a constant?

Reply:
I don't think that was ever implemented in azd, or Aspire.

The way we do it today is we just put the constant directly where it needs to go. i.e. it isn't a "parameter".

Reply:
8.1

Reply:
What we could do is expand the the manifest so you can do this:

```json
{
  "inputs": {
    "value": {
      "type": "string",
      "default": {
        "value": "postgres"
      }
    }
  }
}
```

This would be a trigger for an `aspire-8.1.json` schema. If `value` is provided then `generate` isn't allowed. `azd` would prompt and show the default value (the user just hits ENTER). If it is secret it wouldn't show the value but indicate somehow that it should accept the default.

/cc @vhvb1989

Reply:
For parameters with a default value, azd does not prompt the user for a value.  The strategy is usually just to tell folks what env-var they can set to override the default value.

In your example above, azd would create a bicep param like

```bicep
param postgres-username string = "postgres"
```

then, azd will also create a mapping to an env-var `AZURE_POSTGRES_USERNAME` (I think we do camelCase to SNAKE_UPPER) which users can set (either with `azd env set AZURE_POSTGRES_USERNAME otherValue` or by setting the env var to their system (or a variable or secret on CI).

This strategy allows the overriding w/o too much prompting.  And the mapping to env-vars is already created for all aspire parameters (even for the auto-geneated, IIRC).

Reply:
So do we have enough to decide what we want to go ahead with this? We would need:

1. A change to AZD to support using default values if they are present (in accordance with the schema above).
2. A change to our schema (introduce an 8.1 version of the schema)
3. Introduce the necessary API changes to allow for setting the default value.

Reply:
Can we make this change in a non breaking way that doesn't require a schema update? AZD can prompt in older versions.

Reply:
Well the value field is additive, so we can add it without rev'ing the schema. And tools use use the value if they understand it. We might be able to get away with that?

Reply:
Added an API proposal with the manifest schema change above https://github.com/dotnet/aspire/issues/4429

Reply:
I now think we shouldn't version the schema here. On an older azd you'll get the prompting behavior.

Reply:
We're adding the schema for default values here https://github.com/dotnet/aspire/pull/5586

cc @davidebbo

Reply:
Yes, this seems to describe exactly what we have in that PR. What's a bit more vague is exactly how azd will behave, prompting wise.

> If it is secret it wouldn't show the value but indicate somehow that it should accept the default.

@mitchdenny this part is maybe questionable (iiuc), as it suggests that the secret would be stored in the manifest, but just not shown to the user at prompt time. It's best to say that secrets can never be in the manifest, which relates to https://github.com/dotnet/aspire/issues/5237.